src/Catel.MVVM/Catel.MVVM.Xamarin.Forms/Windows/Controls/ContentPage.cs

```diff
@@ -8,148 +8,196 @@
 namespace Catel.Windows.Controls
 {
     using System;
-
-    using Catel.IoC;
+    using System.ComponentModel;
+    using System.Threading.Tasks;
     using Catel.MVVM;
+    using Catel.MVVM.Providers;
     using Catel.MVVM.Views;
+    using Catel.Threading;
 
     /// <summary>
-    /// The content page.
+    ///     The content page.
     /// </summary>
     public class ContentPage : global::Xamarin.Forms.ContentPage, IView
     {
         /// <summary>
-        /// The view mananger.
+        ///     The view mananger.
         /// </summary>
-        private readonly IViewManager _viewManager;
+        // private readonly IViewManager _viewManager;
+        private readonly UserControlLogic _userControlLogic;
+
+        /// <summary>
+        ///     Initializes a new instance of the <see cref="ContentPage" /> class.
+        /// </summary>
+        public ContentPage() : this(null)
+        {
+            BindingContextChanged += OnBindingContextChanged;
+        }
+
+        private object _oldbindingContext;
+
+        private void OnBindingContextChanged(object o, EventArgs eventArgs)
+        {
+            DataContextChanged.SafeInvoke(this, () => new DataContextChangedEventArgs(_oldbindingContext, BindingContext));
+            _oldbindingContext = BindingContext;
+            RaiseViewModelChanged();
+        }
 
         /// <summary>
-        /// Initializes a new instance of the <see cref="ContentPage"/> class.
+        ///     Initializes a new instance of the <see cref="ContentPage" /> class.
         /// </summary>
-        public ContentPage()
+        /// <param name="viewModel">The view model</param>
+        public ContentPage(IViewModel viewModel)
         {
-            var dependencyResolver = this.GetDependencyResolver();
-            _viewManager = dependencyResolver.Resolve<IViewManager>();
+            _userControlLogic = new UserControlLogic(this, null, viewModel);
+
+            _userControlLogic.TargetViewPropertyChanged += (sender, e) =>
+            {
+                OnPropertyChanged(e.PropertyName);
+            };
+
+            _userControlLogic.ViewModelClosedAsync += OnViewModelClosedAsync;
+
+            _userControlLogic.ViewModelChanged += (sender, args) =>
+            {
+                if (BindingContext != _userControlLogic.ViewModel)

```

Comment:
ObjectHelper.AreEqual (because object might not implement all equality members).